### PR TITLE
add load vault option

### PR DESCRIPTION
**Pull request by anncatton**
Adding vault option for managing secrets. Needed for cron jobs using this script to access protected apis
Implemented for https://github.com/icgc-argo/dac-api/issues/239 and https://github.com/icgc-argo/dac-api/issues/237

### CODE REVIEW

**Review by joneubank — CHANGES_REQUESTED** [review]
Looks like it works. Now we need to update the README to explain:
A - what vault does for us
B - document the env variables
C - How the values should be stored in vault for this to work (they need specific names)

File: README.md [hunk 1]
```diff
@@ -22,6 +22,18 @@ EGO_CLIENT_ID=my-ego-application
...
+
+Vault can be used to store your Ego credentials in a secure manner. When Vault is enabled, you will need to provide these env variables:
+```
+VAULT_URL=https://vault.my-vault-instance.com
+VAULT_ROLE=my-policy-name
+VAULT_SECRETS_PATH=/secrets-dir/path-to-secret/my-secret-name
+```
+Your Ego credential keys stored in your secret should match those listed in Ego Authorization.
+
+`VAULT_TOKEN` is optional and only used to login to Vault for local testing.
```
Comment (joneubank) [inline README.md]: Can we call this out as "should not be used in production", "this is available as a tool for developers to test out a configuration only".

File: README.md [hunk 2]
```diff
@@ -22,6 +22,18 @@ EGO_CLIENT_ID=my-ego-application
...
 
+## Vault
+
+Vault can be used to store your Ego credentials in a secure manner. When Vault is enabled, you will need to provide these env variables:
+```
+VAULT_URL=https://vault.my-vault-instance.com
+VAULT_ROLE=my-policy-name
+VAULT_SECRETS_PATH=/secrets-dir/path-to-secret/my-secret-name
+```
+Your Ego credential keys stored in your secret should match those listed in Ego Authorization.
```
Comment (joneubank) [inline README.md]: This isn't clear to me, but I think you are trying to say:
"The keys of the secrets stored in this vault path need to match the Ego environment variable names: `EGO_URL`, `EGO_CLIENT_ID`, and `EGO_CLIENT_SECRET`"

Comment (anncatton) [reply README.md]: I should be even more specific since it's just the app credentials that need to go into vault, ego url can go in the config

**Review by joneubank — APPROVED** [review]

**Pull request merged**